pkg/artifacts/artifacts.go

```diff
@@ -0,0 +1,77 @@
+package artifacts
+
+import (
+	"os"
+
+	"gopkg.in/yaml.v2"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"github.com/aquasecurity/trivy-kubernetes/pkg/k8s"
+)
+
+// Artifact holds information for kubernetes scannable resources
+type Artifact struct {
+	Namespace string
+	Kind      string
+	Name      string
+	Images    []string
+	resource  map[string]interface{}
+}
+
+// WriteToFile the resource to a file
+func (a *Artifact) WriteToFile(file *os.File) error {
+	data, err := yaml.Marshal(a.resource)
+	if err != nil {
+		return err
+	}
+
+	_, err = file.Write(data)
+	return err
+}
+
+// FromResource is a factory method to create an Artifact from an unstructured.Unstructured
+func FromResource(resource unstructured.Unstructured) (*Artifact, error) {
```

Comment:
I assume this method is relevant for vulnerability scanners only

Reply:
if we change the code to have `iac` and `vuln` as two separate things yes, but for now it will return an artifact with the proper name from metadata, even when it doesn't have an image.